agent/consul/acl_test.go

```diff
@@ -1010,7 +1010,7 @@ func TestACLResolver_Client(t *testing.T) {
 		require.NoError(t, res2.err)
 		require.Equal(t, res1.authz, res2.authz)
 		require.Equal(t, int32(2), tokenReads)
-		require.Equal(t, int32(1), policyResolves)
+		require.Equal(t, int32(2), policyResolves)
```

Comment:
The test needs to be changed to reflect that now the shared policy is resolved twice because of the intentional change to the singleflighting key.

Reply:
You might just want to delete this test or try to force a similar condition.

These concurrency tests are pretty finicky too. The premise of this one was that two tokens being resolved to the same policies and would only result in one `ACL.PolicyResolve` RPC. Given your changes to keying the single flight RPC off of the token instead of the policy IDs, I don't think this test is really valid any longer.

If you did want to keep the test you probably would want to modify it to resolve the same token twice. So it should do 1 token read and 1 policy resolve. With that being said these tests are pretty flaky. Now that the singleflighting is handled by an external module we could argue that testing that singleflighting is probably unnecessary.

Reply:
> If you did want to keep the test you probably would want to modify it to resolve the same token twice. 

I think that's the preceding subtest: `Concurrent-Token-Resolve`.

There may still be value to ensure that the 3rdparty lib is being used correctly to achieve the stated goal, but that can probably be handled by the `Concurrent-Token-Resolve` test. I'll delete the `Concurrent-Policy-Resolve` test.

Reply:
I nuked the concurrent tests and created new tests to cover some of the behavior that was indirectly tested by the concurrency tests. Namely:
* policy resolve returns token-not-found and correctly caches not-found
* policy resolve returns perm denied and correctly deletes the token from the cache completely